modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -168,6 +168,15 @@ sealed abstract class Json extends Product with Serializable {
       case _ => that
     }
 
+  /**
+   *Equality method that does not consider field order or null
+   *values.
+   */
+  def canonicallyEqual(that: Any): Boolean = that match {
+    case that: Json => this.dropNullValues.equals(that.dropNullValues)
```

Comment:
I think "deepDropNullValues" would make more sense here, WDYT?

Reply:
Agreed, but at the same time I think we can make a more memory efficient method for doing the equality check instead of building a entirely new json object sans `NULL` to just throw it away.